### PR TITLE
Enable an optional healthcheck endpoint

### PR DESCRIPTION
**Pull request by jkongie**

### CODE REVIEW

File: server_test.go [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+	)
+
+	// Start serving
+	go s.Serve(lis)
+	defer s.Stop()
+
+	time.Sleep(1 * time.Second)
+
+	// Run a http calll
+	resp, err := http.Get("http://127.0.0.1:1337/healthz")
```
Comment (MStreet3) [inline server_test.go:37]: does it matter that the ports are different between the listener and the server?

Comment (MStreet3) [reply server_test.go:37]: is the sleep to wait for the server to be ready to receive requests?

Comment (jkongie) [reply server_test.go:37]: No the sleep needs to be removed. It was added for debugging

Comment (jkongie) [reply server_test.go:37]: Ports are going to be different anyway. Think 80 http and 442 for https

**Review by MStreet3 — APPROVED** [review]

**Pull request merged**